tests/unittests/core/worker/test_experiment.py

```diff
@@ -643,17 +643,32 @@ def test_status_is_pending_when_increase_max_trials(self, exp_config):
 
     def test_no_increment_when_child_exist(self, create_db_instance):
         """Check that experiment is not incremented when asked for v1 while v2 exists."""
-        query = dict(name='experiment_test', metadata={'user': 'corneauf'}, version=1)
-        get_storage().create_experiment(query)
+        user_args = ['--x~normal(0,1)']
+        metadata = dict(user='tsirif', datetime=datetime.datetime.utcnow(), user_args=user_args)
+        algorithms = {'random': {'seed': None}}
+        config = dict(name='experiment_test', metadata=metadata, version=1, algorithms=algorithms)
 
-        query['version'] = 2
-        query.pop('_id')
+        get_storage().create_experiment(config)
+        original = Experiment('experiment_test', version=1)
 
-        get_storage().create_experiment(query)
+        config['version'] = 2
+        config['metadata']['user_args'].append("--y~+normal(0,1)")
+        config.pop('_id')
 
+        get_storage().create_experiment(config)
+        config.pop('_id')
+
+        config['branch'] = ['experiment_2']
```

Comment:
Why do you change the name? This clearly should override with version 1.

Reply:
Well, the behavior occurs when we use `--branch`, so I thought it be more explicit this way. This also mimics the functional test inside `test_branching` testing the same thing.

Reply:
The problem is version is hard-set to 1 if we change the name. Actually, if we try to run version one but change it without specifying the name, it should raise an error if version 2 already exists right?

Reply:
Yes, but here we're trying to branch from version 1 so we must change the name and explicitly ask for version 1.

Reply:
What happens if we don't?

Reply:
Then, there'll be no `version` key inside the `config` dictionary, so `None` will be passed and the latest experiment will be loaded instead of the first one, basically cancelling the effort we've made to not have that in the first place!

Reply:
I meant if we don't change the name of the experiment, but specify version 1 (even thought there is already a version 2).

Reply:
Then that's a whole other situation which is not what this test is about.

Reply:
The test we need should test for the situation where the modification at line https://github.com/Epistimio/orion/pull/245/files#diff-6d901bae3fe0c9e74e137277f6fdf9b4R578 was required. Since changing the name forces an overwrite of the version number in the config [here](https://github.com/Epistimio/orion/blob/develop/src/orion/core/evc/conflicts.py#L1586) I hardly see of this test would fail if the previous modification was not there. From what I see with a second look at Experiment.configure, the problem will happen when we instantiate with version=1 while a child exists but we don't branch. In this case the version number in the copied experiment will be 2 while it should have been 1.